src/resolver.rs

```diff
@@ -2506,37 +2506,63 @@ impl<'a> ResolveReport<'a> {
                 .push(s);
         }
 
-        let mut last_suggest = vec![];
-        for suggestion in &suggestions {
-            let package = self.graph.nodes[suggestion.package].name.to_string();
-            let command = if suggestion.suggested_diff.from == ROOT_VERSION {
-                FetchCommand::Inspect {
-                    package,
-                    version: suggestion.suggested_diff.to.clone(),
-                }
-            } else {
-                FetchCommand::Diff {
-                    package,
-                    version1: suggestion.suggested_diff.from.clone(),
-                    version2: suggestion.suggested_diff.to.clone(),
-                }
-            };
-            let criteria = self
-                .criteria_mapper
-                .all_criteria_names(&suggestion.suggested_criteria)
-                .map(CriteriaName::from)
-                .collect::<Vec<_>>();
-            last_suggest.push(SuggestedAudit { command, criteria });
-        }
-        cache.set_last_suggest(last_suggest);
-
         Ok(Some(Suggest {
             suggestions,
             suggestions_by_criteria,
             total_lines,
         }))
     }
 
+    /// Given a package name and a delta to be certified, determine the set of
+    /// additional criteria for that delta/version pair which would have a
+    /// healing impact on the audit graph.
+    ///
+    /// This is more reliable than running a suggest and looking for a matching
+    /// output, as it will also select criteria for non-suggested audits.
+    pub fn compute_suggested_criteria(
+        &self,
+        package_name: PackageStr<'_>,
+        delta: &Delta,
+    ) -> Vec<CriteriaName> {
+        let fail = if let Conclusion::FailForVet(fail) = &self.conclusion {
+            fail
+        } else {
+            return Vec::new();
+        };
+
+        let mut criteria = self.criteria_mapper.no_criteria();
+
+        // Enumerate over the recorded failures, adding any criteria for this
+        // delta which would connect that package version into the audit graph.
+        for (&failure_idx, audit_failure) in &fail.failures {
+            let package = &self.graph.nodes[failure_idx];
+            if package.name != package_name {
+                continue;
+            }
+
+            let result = &self.results[failure_idx];
+            for criteria_idx in audit_failure.criteria_failures.all().indices() {
+                let search_result = &result.search_results[criteria_idx];
+                if let SearchResult::Disconnected {
+                    reachable_from_root,
+                    reachable_from_target,
+                } = search_result
+                {
+                    if reachable_from_target.contains(&delta.to)
+                        && (delta.from == ROOT_VERSION || reachable_from_root.contains(&delta.from))
+                    {
```

Comment:
Do we not add ROOT_VERSION to its own reachability set??? (The first part of the `||` seems pointless)

Reply:
I don't know, I just added in the explicit case to be extra sure. I can drop it and we can see if it still works I suppose